### PR TITLE
refactor(game): begin refactoring into a game state and game level pa…

### PR DESCRIPTION
**Pull request by seiyria**
…ttern

### CODE REVIEW

File: app/js/gamemodes/valet.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      for (let i = gamestate.cars.length - 1; i >= 0; i--) {
+        gamestate.cars[i].update()
+      }
+    }
+
+    requestAnimationFrame(Valet.loop);
+    if (gamestate.playing) config.world.step(1/60);
+    config.renderer.render(config.stage);
+  }
+}
```
Comment (Fishrock123) [inline app/js/gamemodes/valet.js:53]: EOF newlines please

Comment (seiyria) [inline app/js/gamemodes/valet.js:53]: why

**Comment by Fishrock123** [comment]
Could we name Class-exporting files by caps like we've been doing?

**Comment by Fishrock123** [comment]
also uh, commit message length

**Comment by seiyria** [comment]
Is that how they're capitalized? It seemed mostly random to me. I would prefer if either all files are capitalized or none of them are because that's a mostly-useless notation if the file structure makes sense (currently everything is just in one folder, so this will require a bit of a re-org).

Also, if we want to enforce commit message length we should probably have a bot for that (gitcop is ok).

**Comment by Fishrock123** [comment]
Just fix your vim or whatever you commit with so that it enforces 50colunns message, 72 columns description

**Comment by seiyria** [comment]
I use the terminal to commit, but you haven't given my a reason why I should care about commit message length. I don't do things without a sufficient reason why.

**Comment by Fishrock123** [comment]
It cuts off the description in git. Describe what you can in the message and then dump the rest in the description. Be concise.

Also >72 lines description wraps badly in terminals.

**Comment by seiyria** [comment]
Make your terminal larger ????

idk, I'm not rebasing or changing messages but we'll see going forward.

**Comment by Fishrock123** [comment]
Fine I'll just fix your commits for you when I get home.

Also git does it regardless of your terminal size.

**Pull request merged**